application/classes/Ushahidi/Repository/Form/Contact.php

```diff
@@ -203,6 +217,45 @@ public function existsInFormContactByContactNumber($contact, $form_id)
 			->get('total');
 	}
 
+	/**
+	 * @param int $contact_id
+	 * @param int $form_id
+	 * @return bool
+	 *
+	 */
+	public function existsInActiveTargetedSurveyByContactNumber($contact)
+	{
+		$where = array(
+			'contacts.contact' => $contact,
+			'targeted_survey_state.survey_status' => array('PENDING RESPONSE', 'RECEIVED RESPONSE')
+		);
+		$query = $this->selectQuery($where)
+			->resetSelect()
+			->select(
+				['targeted_survey_state.id', 'targeted_survey_state_id'], ['contacts.contact', 'contact'], ['targeted_survey_state.contact_id', 'contact_id'], ['targeted_survey_state.form_id', 'form_id']
+			)
+			->limit(1);
+		$query = $this->targetedSurveyStateJoin($query);
+		$result = $query
+			->execute($this->db);
+		if ($result) {
+			return $result->current();
+		}
+	}
+
+
+	/**
+	 * @param int $contact_id
+	 * @param int $form_id
+	 * @return bool
+	 */
+	public function setInactiveTargetedSurvey($tss_id)
+	{
+		$repo = $this->targeted_survey_state_repo->get($tss_id);
+		$entity = $repo->setState(array('survey_status' => 'INVALID'));
```

Comment:
do we currently depend on this to say 'INVALID', or can it be switched to something like 'REMOVED_FROM_SURVEY' or 'MOVED_TO_SURVEY_XXXX' if we know what that new survey is?

Reply:
mmmm. I think that might be a good idea yea. Let me check with Will. I also will hold on this merge because I want to move statuses to consts. This hardcoded statuses without even a const are gross